### PR TITLE
chore: resolution config updates

### PR DESCRIPTION
**Pull request by enaqx**
- Update Readme
- Update default provider configs from Infura to Alchemy
- Change rinkeby network to goerli

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,7 +64,7 @@ func main() {
   }
 
   // Set custom Ethereum endpoint for UNS service
-  ethContractBackend, _ := ethclient.Dial("https://mainnet.infura.io/v3/c5da69dfac9c4d9d96dd232580d4124e")
+  ethContractBackend, _ := ethclient.Dial("https://eth-mainnet.alchemyapi.io/v2/RAQcwz7hhKhmwgoti6HYM_M_9nRJjEsQ")
```
Comment (sammyluo) [inline README.md:67]: Let's keep it consistent with other alchemy url
```suggestion
  ethContractBackend, _ := ethclient.Dial("https://eth-mainnet.g.alchemy.com/v2/RAQcwz7hhKhmwgoti6HYM_M_9nRJjEsQ")
```

File: README.md [hunk 2]
```diff
@@ -101,9 +88,9 @@ import (
 )
 
 func main() {
-  var infuraApiKey = INFURA_PROJECT_ID
-  var ethereumUrl = "https://mainnet.infura.io/v3/" + infuraApiKey
-  var ethereumL2Url = "https://polygon-mumbai.infura.io/v3/" + infuraApiKey
+  var alchemyApiKey = ALCHEMY_PROJECT_ID
+  var ethereumUrl = "https://eth-mainnet.alchemyapi.io/v2/" + alchemyApiKey
```
Comment (sammyluo) [inline README.md:92]:
```suggestion
  var ethereumUrl = "https://eth-mainnet.g.alchemy.com/v2/" + alchemyApiKey
```

File: uns_builder_test.go [hunk 3]
```diff
@@ -26,9 +26,9 @@ func TestUnsBuilder(t *testing.T) {
 
 func TestUnsBuilderSetBackend(t *testing.T) {
 	t.Parallel()
-	backendl1, _ := ethclient.Dial("https://rinkeby.infura.io/v3/c5da69dfac9c4d9d96dd232580d4124e")
-	backendl2, _ := ethclient.Dial("https://matic-testnet-archive-rpc.bwarelabs.com/")
-	builder := NewUnsBuilder().SetEthereumNetwork("rinkeby").SetL2EthereumNetwork("mumbai")
+	backendl1, _ := ethclient.Dial("https://eth-goerli.alchemyapi.io/v2/r3soltxtCL_KhZGl7EwWsWax5ll_MFTN")
```
Comment (sammyluo) [inline uns_builder_test.go:29]:
```suggestion
	backendl1, _ := ethclient.Dial("https://eth-goerli.g.alchemy.com/v2/r3soltxtCL_KhZGl7EwWsWax5ll_MFTN")
```

File: uns_builder_test.go [hunk 4]
```diff
@@ -69,8 +69,8 @@ func TestUnsBuilderChecksL2Network(t *testing.T) {
 func TestUnsBuilderChecksL2ContractBackend(t *testing.T) {
 	t.Parallel()
 	var expectedError *UnsConfigurationError
-	backendl1, _ := ethclient.Dial("https://rinkeby.infura.io/v3/c5da69dfac9c4d9d96dd232580d4124e")
-	builder := NewUnsBuilder().SetEthereumNetwork("mainnet").SetL2EthereumNetwork("mumbai")
+	backendl1, _ := ethclient.Dial("https://eth-goerli.alchemyapi.io/v2/r3soltxtCL_KhZGl7EwWsWax5ll_MFTN")
```
Comment (sammyluo) [inline uns_builder_test.go:72]:
```suggestion
	backendl1, _ := ethclient.Dial("https://eth-goerli.g.alchemy.com/v2/r3soltxtCL_KhZGl7EwWsWax5ll_MFTN")
```

File: uns_builder_test.go [hunk 5]
```diff
@@ -80,8 +80,8 @@ func TestUnsBuilderChecksL2ContractBackend(t *testing.T) {
 func TestUnsBuilderChecksL1ContractBackend(t *testing.T) {
 	t.Parallel()
 	var expectedError *UnsConfigurationError
-	backendl2, _ := ethclient.Dial("https://rinkeby.infura.io/v3/c5da69dfac9c4d9d96dd232580d4124e")
-	builder := NewUnsBuilder().SetEthereumNetwork("mainnet").SetL2EthereumNetwork("mumbai")
+	backendl2, _ := ethclient.Dial("https://eth-goerli.alchemyapi.io/v2/r3soltxtCL_KhZGl7EwWsWax5ll_MFTN")
```
Comment (sammyluo) [inline uns_builder_test.go:83]:
```suggestion
	backendl2, _ := ethclient.Dial("https://eth-goerli.g.alchemy.com/v2/r3soltxtCL_KhZGl7EwWsWax5ll_MFTN")
```

File: uns_builder_test.go [hunk 6]
```diff
@@ -26,9 +26,9 @@ func TestUnsBuilder(t *testing.T) {
 
 func TestUnsBuilderSetBackend(t *testing.T) {
 	t.Parallel()
-	backendl1, _ := ethclient.Dial("https://rinkeby.infura.io/v3/c5da69dfac9c4d9d96dd232580d4124e")
-	backendl2, _ := ethclient.Dial("https://matic-testnet-archive-rpc.bwarelabs.com/")
-	builder := NewUnsBuilder().SetEthereumNetwork("rinkeby").SetL2EthereumNetwork("mumbai")
+	backendl1, _ := ethclient.Dial("https://eth-goerli.alchemyapi.io/v2/r3soltxtCL_KhZGl7EwWsWax5ll_MFTN")
+	backendl2, _ := ethclient.Dial("https://polygon-mumbai.g.alchemy.com/v2/EP3SMW2f-2FMABuWuEdHdxKq_v1_ww8")
```
Comment (sammyluo) [inline uns_builder_test.go:30]: Typo in the key `EP3SMW2f-2FMABuWuEdHdxKq_v1_ww82` (missing the ending `2`)

```suggestion
	backendl2, _ := ethclient.Dial("https://polygon-mumbai.g.alchemy.com/v2/EP3SMW2f-2FMABuWuEdHdxKq_v1_ww82")
```

**Review by sammyluo — CHANGES_REQUESTED** [review]

**Comment by enaqx** [comment]
@sammyluo Updated, please take a look.

**Review by sammyluo — APPROVED** [review]
LGTM! 🙌

**Pull request merged**